### PR TITLE
component-setup-refactor-routes-4

### PR DESCRIPTION
**Pull request by ramagj**

### CODE REVIEW

File: src/app/views/forums/landing/active/RouteForumsLandingActive.vue [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	}),
+};
+</script>
+
+<script lang="ts" setup>
+const topics = ref<ForumTopicModel[]>([]);
+const postCountPerPage = ref(0);
+
+createAppRoute({
+	routeTitle: computed(() => $gettext(`Active Topics in All Forums`)),
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/forums/landing/active/RouteForumsLandingActive.vue:26]: We usually put gettext in computed or toref actually. Not really required, but the original thinking was that you'd be able to swap languages fast then. We still force reload the page after changing language, so it doesn't really matter, but the habit stuck.

File: src/app/views/landing/game-api-doc/RouteLandingGameApiDoc.vue [hunk 2]
```diff
@@ -1,35 +1,32 @@
...
-	OptionsForLegacyRoute,
-} from '../../../../_common/route/legacy-route-component';
+import { computed } from 'vue';
+import { RouterView, useRoute } from 'vue-router';
+import { createAppRoute, defineAppRouteOptions } from '../../../../_common/route/route-component';
 import AppThemeSvg from '../../../../_common/theme/svg/AppThemeSvg.vue';
+import { $gettext } from '../../../../_common/translate/translate.service';
+
+// TODO(component-setup-refactor-routes-4): do we need to add 'allowSyntheticDefaultImports' flag to tsconfig.json?
 import nav from '../../../../lib/doc-game-api/v1.x/nav.json';
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/landing/game-api-doc/RouteLandingGameApiDoc.vue:9]: Are the routes working? If they work, we don't need to do anything, yeah?

File: src/app/views/landing/marketplace/RouteLandingMarketplace.vue [hunk 3]
```diff
@@ -1,62 +1,47 @@
...
+createAppRoute({
+	routeTitle: computed(() => $gettext(`Sell Your Games`)),
+	onResolved({ payload }) {
+		Meta.description = payload.metaDescription;
+		Meta.fb = payload.fb;
+		Meta.twitter = payload.twitter;
+		Meta.fb.image = Meta.twitter.image = socialImage;
+		// TODO(component-setup-refactor-routes-4): Seems like not being used elsewhere, can be removed?
+		firesidePosts = FiresidePostModel.populate(payload.firesidePosts);
+		games = GameModel.populate(payload.games);
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/landing/marketplace/RouteLandingMarketplace.vue:42]: yeah, remove anything not used

File: src/app/views/legal/ads/RouteLegalAds.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+import { computed } from 'vue';
+import { createAppRoute, defineAppRouteOptions } from '../../../../_common/route/route-component';
+import { $gettext } from '../../../../_common/translate/translate.service';
+
+// TODO(component-setup-refactor-routes-4): html not available in this import?
+import { html } from '../../../../lib/terms/ads/global.md';
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/legal/ads/RouteLegalAds.vue:7]: should be fine to ignore, the old components also complained

File: src/app/views/legal/cookies/RouteLegalCookies.vue [hunk 5]
```diff
@@ -0,0 +1,55 @@
+<script lang="ts">
+import { computed } from 'vue';
+import { createAppRoute, defineAppRouteOptions } from '../../../../_common/route/route-component';
+import { $gettext } from '../../../../_common/translate/translate.service';
+
+// TODO(component-setup-refactor-routes-4): html not available in this import?
+import { html } from '../../../../lib/terms/cookies/global.md';
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/legal/cookies/RouteLegalCookies.vue:7]: should be fine to ignore, the old components also complained

File: src/app/views/legal/deletion/RouteLegalDeletion.vue [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<script lang="ts">
+import { computed } from 'vue';
+import { createAppRoute, defineAppRouteOptions } from '../../../../_common/route/route-component';
+import { $gettext } from '../../../../_common/translate/translate.service';
+// TODO(component-setup-refactor-routes-4): html not available in this import?
+import { html } from '../../../../lib/terms/deletion/global.md';
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/legal/deletion/RouteLegalDeletion.vue:6]: should be fine to ignore, the old components also complained

File: src/app/views/legal/cookies/RouteLegalCookies.vue [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	const expires = 'expires=' + d.toUTCString();
+	document.cookie = cname + '=' + cvalue + ';' + expires + ';path=/';
+}
+
+function playwireOptOut() {
+	playwireSetCookie('_pubcid_optout', '1', 1825);
+
+	// TODO(component-setup-refactor-routes-4): do we need this TODO still?
+	// TODO: should probably show some confirmation, yeah?
+	// At this point they should be opted out.
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/legal/cookies/RouteLegalCookies.vue:29]: You can leave the todo in for now. I actually have to update that entirely if we switch over to the new ad network anyway. Plus, we don't even use playwire at all anymore.

File: src/app/views/legal/privacy/RouteLegalPrivacy.vue [hunk 8]
```diff
@@ -1,40 +1,40 @@
...
-import {
-	LegacyRouteComponent,
-	OptionsForLegacyRoute,
-} from '../../../../_common/route/legacy-route-component';
+import { computed } from 'vue';
+import { createAppRoute, defineAppRouteOptions } from '../../../../_common/route/route-component';
+import { $gettext } from '../../../../_common/translate/translate.service';
+
+// TODO(component-setup-refactor-routes-4): html not available in this import?
 import { html } from '../../../../lib/terms/privacy/global.md';
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/legal/privacy/RouteLegalPrivacy.vue:7]: same as above

File: src/app/views/legal/terms/RouteLegalTerms.vue [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<script lang="ts">
+import { computed } from 'vue';
+import { createAppRoute, defineAppRouteOptions } from '../../../../_common/route/route-component';
+import { $gettext } from '../../../../_common/translate/translate.service';
+
+// TODO(component-setup-refactor-routes-4): html not available in this import?
+import { html } from '../../../../lib/terms/terms/global.md';
```
[the remark this answers is not in the document]
Comment (hworld) [reply src/app/views/legal/terms/RouteLegalTerms.vue:7]: same

File: src/app/views/landing/redlight/RouteLandingRedlight.vue [hunk 10]
```diff
@@ -169,6 +163,7 @@ export default class RouteLandingRedlight extends LegacyRouteComponent {
 								<b>
 									Create a Game Jolt account yourself directly. No need for an
 									account manager.
+									<br />
```
[the remark this answers is not in the document]
Comment (ramagj) [reply src/app/views/landing/redlight/RouteLandingRedlight.vue:166]: No, I wrapped everything with gettext() and added a manual break. Missed removing this one

**Pull request merged**